{"ast":null,"code":"import axios from 'axios';\n\n// const usersUrl = 'http://localhost:3003/users';\nconst usersUrl = 'http://localhost:8000';\nexport const StudentDetail = async (RollNo, formData) => {\n  try {\n    const response = await axios.post(`${usersUrl}/${RollNo}/StudentDetail`, formData);\n    return response;\n  } catch (error) {\n    console.error('Error fetching users Calling on StudentDetail API:', error);\n    throw error; // Re-throw the error so it can be caught by the caller\n  }\n};\n_c = StudentDetail;\nexport const StudentAddress = async (RollNo, formData) => {\n  try {\n    const response = await axios.post(`${usersUrl}/${RollNo}/StudentAddress`, formData);\n    return response;\n  } catch (error) {\n    console.error('Error fetching users Calling on StudentAddress API:', error);\n    throw error; // Re-throw the error so it can be caught by the caller\n  }\n};\n_c2 = StudentAddress;\nexport const SemesterMarks = async (RollNo, formData) => {\n  try {\n    const response = await axios.post(`${usersUrl}/${RollNo}/SemesterMarks`, formData);\n    return response;\n  } catch (error) {\n    console.error('Error fetching users Calling on SemesterMarks API:', error);\n    throw error; // Re-throw the error so it can be caught by the caller\n  }\n};\n_c3 = SemesterMarks;\nexport const SubjectDetails = async (RollNo, formData) => {\n  try {\n    const response = await axios.post(`${usersUrl}/${RollNo}/SubjectDetails`, formData);\n    return response;\n  } catch (error) {\n    console.error('Error fetching users Calling on SemesterMarks API:', error);\n    throw error; // Re-throw the error so it can be caught by the caller\n  }\n};\n_c4 = SubjectDetails;\nexport const SemesterSubject = async (RollNo, formData) => {\n  try {\n    const response = await axios.get(`${usersUrl}/SemesterSubject`, formData);\n    return response;\n  } catch (error) {\n    console.error('Error fetching users Calling on SemesterMarks API:', error);\n    throw error; // Re-throw the error so it can be caught by the caller\n  }\n};\n// export const SignInUser = async (user) => {\n//   try {\n//     const response = await axios.post(`${usersUrl}/signin`, user);\n//     return response;\n//   } catch (error) {\n//     console.error('Error fetching users Calling on SignIN API:', error);\n//     throw error; // Re-throw the error so it can be caught by the caller\n//   }\n// };\n\n// export const getUsers = async (month,year,filter) => {\n//     try {\n//       const response = await axios.get(`${usersUrl}/all/admin/${year}/${month}`); return response;\n//     } catch (error) {\n//       console.error('Error fetching users Calling on getUsers API:', error);\n//       throw error; // Re-throw the error so it can be caught by the caller\n//     }\n//   };\n//   export const getSalary = async (id,users) => {\n//     console.log(users);\n//     try {\n//       const response = await axios.post(`${usersUrl}/staffSalary/staffDash/${id}`,users); return response;\n//     } catch (error) {\n//       console.error('Error fetching users Calling on getSalary API:', error);\n//       throw error; // Re-throw the error so it can be caught by the caller\n//     }\n//   };\n//   export const SearchStaff = async (user) => {\n//     try {\n//       const response = await axios.get(`${usersUrl}/search`, {\n//         params: user,\n//       });\n//       return response;\n//     } catch (error) {\n//       console.error('Error fetching users:', error);\n//       throw error; // Re-throw the error so it can be caught by the caller\n//     }\n//   };\n// export const addUser = async (year,month,user) => {\n//     return await axios.post(`${usersUrl}/data-entry/${year}/${month}`, user);\n// }\n\n// export const deleteUser = async (id,month,year) => {\n//     return await axios.delete(`${usersUrl}/${id}/${month}/${year}`);\n// }\n// export const getUser = async (id, month,year) => {\n//     try {\n//         return await axios.get(`${usersUrl}/${id}/${year}/${month}`);\n//     } catch (error) {\n//         console.error(\"Error calling getUser api in api.js\", error);\n//         throw error; // Rethrow the error to handle it in the component\n//     }\n// }\n\n// export const EditUser = async (id,month,year, user) => {\n//     return await axios.put(`${usersUrl}/${id}/${year}/${month}`, user)\n// }\n\n// export const resetPassword = async (email) => {\n//   try{\n//   return await axios.put(`${usersUrl}/forgot-password`, email);\n//   }\n//   catch(error){\n//     console.log(\"Error on calling resetPassword\",error);\n//   }\n// }\n// // export const getStudent =async(user)=>{\n// //     return await axios.post(`${usersUrl}/Student`,user)\n// // }\n\n// export const fetchData = async (wing, month,year) => {\n//   try {\n//       return await axios.get(`${usersUrl}/${wing}/${year}/${month}`);\n//   } catch (error) {\n//       console.error(\"Error calling getUser api in api.js\", error);\n//       throw error; // Rethrow the error to handle it in the component\n//   }\n// }\n_c5 = SemesterSubject;\nvar _c, _c2, _c3, _c4, _c5;\n$RefreshReg$(_c, \"StudentDetail\");\n$RefreshReg$(_c2, \"StudentAddress\");\n$RefreshReg$(_c3, \"SemesterMarks\");\n$RefreshReg$(_c4, \"SubjectDetails\");\n$RefreshReg$(_c5, \"SemesterSubject\");","map":{"version":3,"names":["axios","usersUrl","StudentDetail","RollNo","formData","response","post","error","console","_c","StudentAddress","_c2","SemesterMarks","_c3","SubjectDetails","_c4","SemesterSubject","get","_c5","$RefreshReg$"],"sources":["E:/AithStudentPortal/E-Student-Module/src/Services/api.js"],"sourcesContent":["import axios from 'axios';\r\n\r\n// const usersUrl = 'http://localhost:3003/users';\r\nconst usersUrl = 'http://localhost:8000';\r\n\r\nexport const StudentDetail = async (RollNo,formData) => {\r\n  try {\r\n    const response = await axios.post(`${usersUrl}/${RollNo}/StudentDetail`, formData);\r\n    return response;\r\n  } catch (error) {\r\n    console.error('Error fetching users Calling on StudentDetail API:', error);\r\n    throw error; // Re-throw the error so it can be caught by the caller\r\n  }\r\n};\r\n\r\nexport const StudentAddress = async (RollNo,formData) => {\r\n    try {\r\n      const response = await axios.post(`${usersUrl}/${RollNo}/StudentAddress`, formData);\r\n      return response;\r\n    } catch (error) {\r\n      console.error('Error fetching users Calling on StudentAddress API:', error);\r\n      throw error; // Re-throw the error so it can be caught by the caller\r\n    }\r\n  };\r\n  export const SemesterMarks = async (RollNo,formData) => {\r\n    try {\r\n      const response = await axios.post(`${usersUrl}/${RollNo}/SemesterMarks`, formData);\r\n      return response;\r\n    } catch (error) {\r\n      console.error('Error fetching users Calling on SemesterMarks API:', error);\r\n      throw error; // Re-throw the error so it can be caught by the caller\r\n    }\r\n  };\r\n  export const SubjectDetails = async (RollNo,formData) => {\r\n    try {\r\n      const response = await axios.post(`${usersUrl}/${RollNo}/SubjectDetails`, formData);\r\n      return response;\r\n    } catch (error) {\r\n      console.error('Error fetching users Calling on SemesterMarks API:', error);\r\n      throw error; // Re-throw the error so it can be caught by the caller\r\n    }\r\n  };\r\n  export const SemesterSubject = async (RollNo,formData) => {\r\n    try {\r\n      const response = await axios.get(`${usersUrl}/SemesterSubject`, formData);\r\n      return response;\r\n    } catch (error) {\r\n      console.error('Error fetching users Calling on SemesterMarks API:', error);\r\n      throw error; // Re-throw the error so it can be caught by the caller\r\n    }\r\n  };\r\n// export const SignInUser = async (user) => {\r\n//   try {\r\n//     const response = await axios.post(`${usersUrl}/signin`, user);\r\n//     return response;\r\n//   } catch (error) {\r\n//     console.error('Error fetching users Calling on SignIN API:', error);\r\n//     throw error; // Re-throw the error so it can be caught by the caller\r\n//   }\r\n// };\r\n\r\n// export const getUsers = async (month,year,filter) => {\r\n//     try {\r\n//       const response = await axios.get(`${usersUrl}/all/admin/${year}/${month}`); return response;\r\n//     } catch (error) {\r\n//       console.error('Error fetching users Calling on getUsers API:', error);\r\n//       throw error; // Re-throw the error so it can be caught by the caller\r\n//     }\r\n//   };\r\n//   export const getSalary = async (id,users) => {\r\n//     console.log(users);\r\n//     try {\r\n//       const response = await axios.post(`${usersUrl}/staffSalary/staffDash/${id}`,users); return response;\r\n//     } catch (error) {\r\n//       console.error('Error fetching users Calling on getSalary API:', error);\r\n//       throw error; // Re-throw the error so it can be caught by the caller\r\n//     }\r\n//   };\r\n//   export const SearchStaff = async (user) => {\r\n//     try {\r\n//       const response = await axios.get(`${usersUrl}/search`, {\r\n//         params: user,\r\n//       });\r\n//       return response;\r\n//     } catch (error) {\r\n//       console.error('Error fetching users:', error);\r\n//       throw error; // Re-throw the error so it can be caught by the caller\r\n//     }\r\n//   };\r\n// export const addUser = async (year,month,user) => {\r\n//     return await axios.post(`${usersUrl}/data-entry/${year}/${month}`, user);\r\n// }\r\n\r\n// export const deleteUser = async (id,month,year) => {\r\n//     return await axios.delete(`${usersUrl}/${id}/${month}/${year}`);\r\n// }\r\n// export const getUser = async (id, month,year) => {\r\n//     try {\r\n//         return await axios.get(`${usersUrl}/${id}/${year}/${month}`);\r\n//     } catch (error) {\r\n//         console.error(\"Error calling getUser api in api.js\", error);\r\n//         throw error; // Rethrow the error to handle it in the component\r\n//     }\r\n// }\r\n\r\n// export const EditUser = async (id,month,year, user) => {\r\n//     return await axios.put(`${usersUrl}/${id}/${year}/${month}`, user)\r\n// }\r\n\r\n// export const resetPassword = async (email) => {\r\n//   try{\r\n//   return await axios.put(`${usersUrl}/forgot-password`, email);\r\n//   }\r\n//   catch(error){\r\n//     console.log(\"Error on calling resetPassword\",error);\r\n//   }\r\n// }\r\n// // export const getStudent =async(user)=>{\r\n// //     return await axios.post(`${usersUrl}/Student`,user)\r\n// // }\r\n\r\n// export const fetchData = async (wing, month,year) => {\r\n//   try {\r\n//       return await axios.get(`${usersUrl}/${wing}/${year}/${month}`);\r\n//   } catch (error) {\r\n//       console.error(\"Error calling getUser api in api.js\", error);\r\n//       throw error; // Rethrow the error to handle it in the component\r\n//   }\r\n// }"],"mappings":"AAAA,OAAOA,KAAK,MAAM,OAAO;;AAEzB;AACA,MAAMC,QAAQ,GAAG,uBAAuB;AAExC,OAAO,MAAMC,aAAa,GAAG,MAAAA,CAAOC,MAAM,EAACC,QAAQ,KAAK;EACtD,IAAI;IACF,MAAMC,QAAQ,GAAG,MAAML,KAAK,CAACM,IAAI,CAAE,GAAEL,QAAS,IAAGE,MAAO,gBAAe,EAAEC,QAAQ,CAAC;IAClF,OAAOC,QAAQ;EACjB,CAAC,CAAC,OAAOE,KAAK,EAAE;IACdC,OAAO,CAACD,KAAK,CAAC,oDAAoD,EAAEA,KAAK,CAAC;IAC1E,MAAMA,KAAK,CAAC,CAAC;EACf;AACF,CAAC;AAACE,EAAA,GARWP,aAAa;AAU1B,OAAO,MAAMQ,cAAc,GAAG,MAAAA,CAAOP,MAAM,EAACC,QAAQ,KAAK;EACrD,IAAI;IACF,MAAMC,QAAQ,GAAG,MAAML,KAAK,CAACM,IAAI,CAAE,GAAEL,QAAS,IAAGE,MAAO,iBAAgB,EAAEC,QAAQ,CAAC;IACnF,OAAOC,QAAQ;EACjB,CAAC,CAAC,OAAOE,KAAK,EAAE;IACdC,OAAO,CAACD,KAAK,CAAC,qDAAqD,EAAEA,KAAK,CAAC;IAC3E,MAAMA,KAAK,CAAC,CAAC;EACf;AACF,CAAC;AAACI,GAAA,GARSD,cAAc;AASzB,OAAO,MAAME,aAAa,GAAG,MAAAA,CAAOT,MAAM,EAACC,QAAQ,KAAK;EACtD,IAAI;IACF,MAAMC,QAAQ,GAAG,MAAML,KAAK,CAACM,IAAI,CAAE,GAAEL,QAAS,IAAGE,MAAO,gBAAe,EAAEC,QAAQ,CAAC;IAClF,OAAOC,QAAQ;EACjB,CAAC,CAAC,OAAOE,KAAK,EAAE;IACdC,OAAO,CAACD,KAAK,CAAC,oDAAoD,EAAEA,KAAK,CAAC;IAC1E,MAAMA,KAAK,CAAC,CAAC;EACf;AACF,CAAC;AAACM,GAAA,GARWD,aAAa;AAS1B,OAAO,MAAME,cAAc,GAAG,MAAAA,CAAOX,MAAM,EAACC,QAAQ,KAAK;EACvD,IAAI;IACF,MAAMC,QAAQ,GAAG,MAAML,KAAK,CAACM,IAAI,CAAE,GAAEL,QAAS,IAAGE,MAAO,iBAAgB,EAAEC,QAAQ,CAAC;IACnF,OAAOC,QAAQ;EACjB,CAAC,CAAC,OAAOE,KAAK,EAAE;IACdC,OAAO,CAACD,KAAK,CAAC,oDAAoD,EAAEA,KAAK,CAAC;IAC1E,MAAMA,KAAK,CAAC,CAAC;EACf;AACF,CAAC;AAACQ,GAAA,GARWD,cAAc;AAS3B,OAAO,MAAME,eAAe,GAAG,MAAAA,CAAOb,MAAM,EAACC,QAAQ,KAAK;EACxD,IAAI;IACF,MAAMC,QAAQ,GAAG,MAAML,KAAK,CAACiB,GAAG,CAAE,GAAEhB,QAAS,kBAAiB,EAAEG,QAAQ,CAAC;IACzE,OAAOC,QAAQ;EACjB,CAAC,CAAC,OAAOE,KAAK,EAAE;IACdC,OAAO,CAACD,KAAK,CAAC,oDAAoD,EAAEA,KAAK,CAAC;IAC1E,MAAMA,KAAK,CAAC,CAAC;EACf;AACF,CAAC;AACH;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;;AAEA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;;AAEA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;;AAEA;AACA;AACA;;AAEA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;;AAEA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AAAAW,GAAA,GAtFeF,eAAe;AAAA,IAAAP,EAAA,EAAAE,GAAA,EAAAE,GAAA,EAAAE,GAAA,EAAAG,GAAA;AAAAC,YAAA,CAAAV,EAAA;AAAAU,YAAA,CAAAR,GAAA;AAAAQ,YAAA,CAAAN,GAAA;AAAAM,YAAA,CAAAJ,GAAA;AAAAI,YAAA,CAAAD,GAAA"},"metadata":{},"sourceType":"module","externalDependencies":[]}